nixops_aws/resources/aws_data_lifecycle_manager.py

```diff
@@ -0,0 +1,253 @@
+        args["ExecutionRoleArn"] = self.arn_from_role_name(config["executionRole"])
+        args["Description"] = config["description"]
+        args["State"] = "ENABLED"
+        args["PolicyDetails"] = dict(
+            TargetTags=[
+                {"Key": k, "Value": config["targetTags"][k]}
+                for k in config["targetTags"]
+            ],
+            Schedules=[
+                dict(
+                    Name=config["dlmName"],
+                    CopyTags=config["copyTags"],
+                    TagsToAdd=[
+                        {"Key": k, "Value": config["tagsToAdd"][k]}
+                        for k in config["tagsToAdd"]
+                    ],
+                    CreateRule=dict(
+                        Interval=config["ruleInterval"],
+                        IntervalUnit=config["ruleIntervalUnit"].upper(),
+                        Times=[config["ruleTime"], ],
+                    ),
+                    RetainRule=dict(Count=config["retainRule"]),
+                ),
+            ],
+            Parameters=dict(ExcludeBootVolume=config["excludeBootVolume"]),
+        )
+        with self.depl._db:
+            self.description = config["description"]
+            self.executionRole = config["executionRole"]
+            self.resourceTypes = config["resourceTypes"]
+            self.targetTags = config["targetTags"]
+            self.excludeBootVolume = config["excludeBootVolume"]
+            self.copyTags = config["copyTags"]
+            self.tagsToAdd = config["tagsToAdd"]
+            self.ruleInterval = config["ruleInterval"]
+            self.ruleIntervalUnit = config["ruleIntervalUnit"]
+            self.ruleTime = config["ruleTime"]
+            self.retainRule = config["retainRule"]
+            self.dlmName = config["dlmName"]
+
+        return args
+
+    def create(
+        self,
+        defn: awsDataLifecycleManagerDefinition,
+        check,
+        allow_reboot,
+        allow_recreate,
+    ):
+        config = defn.config
+        if self.region is None:
+            self.region = config.region
+        elif self.region != config.region or (
+            self.resourceTypes != config.resourceTypes
+            and self.resourceTypes is not None
+        ):
+            self.warn(
+                "cannot change region or resource types for an existing data lifecycle manager..."
+            )
+
+        self.access_key_id = config.accessKeyId
+        if self.state == self.UP:
+            dlm_keys = [
+                "copyTags",
+                "retainRule",
+                "ruleTime",
+                "description",
+                "executionRole",
+                "ruleIntervalUnit",
+                "tagsToAdd",
+                "excludeBootVolume",
+                "resourceTypes",
+                "targetTags",
+                "ruleInterval",
+                "dlmName",
+            ]
+            dlm_diff = [k for k in dlm_keys if getattr(self, k) != config[k]]
```

Comment:
@tewfik-ghariani  the only remaining issue is here `error: Value of type "AwsDataLifecycleManagerOptions" is not indexable`

Reply:
config.k should work I guess?

Reply:
or something like "config.get(k)"